trajectory_utils/src/trajectory_utils/conversions/conversions.cpp

```diff
@@ -120,7 +120,33 @@ void time_to_speed(const std::vector<double>& downtrack, const std::vector<doubl
     double dt = cur_time - prev_time;
     double delta_d = cur_pos - prev_position;
 
-    double cur_speed = (2.0 * delta_d / dt) - prev_speed;
+    double cur_speed;
+    double jerk_min = 0.01; //Min stop and wait jerk
+    
+    //if stop and wait, use constant jerk equations
+    if(isStopandWait[i]){
+
+      if(delta_d == 0){
+        cur_speed =0;
+        break;
+      }
+
+      if(decel_jerk > jerk_min){
+        cur_speed = prev_speed - 0.5* decel_jerk*pow(dt,2);
+        if(cur_speed < 0){
+          cur_speed = 0;
+        }
+      }
+      else{
+        // stop and wait plugin doesn't create slow down traj for very low jerk requirement
+        cur_speed = prev_speed;
```

Comment:
I understand that this is meant to mimic how stop and wait works, so this might not be an issue, but strictly speaking 0 jerk does not correlate to constant speed. If a vehicle already has some acceleration it can leave that acceleration unchanged (0 jerk) and still move faster than it did before at each subsequent moment.